pkg/kubelet/im/image_gc.go

```diff
@@ -104,38 +97,57 @@ func newImageManager(runtime container.Runtime, cadvisorInterface cadvisor.Inter
 	if policy.LowThresholdPercent > policy.HighThresholdPercent {
 		return nil, fmt.Errorf("LowThresholdPercent %d can not be higher than HighThresholdPercent %d", policy.LowThresholdPercent, policy.HighThresholdPercent)
 	}
-	im := &realImageManager{
+
+	im := &realImageGC{
 		runtime:      runtime,
 		policy:       policy,
+		minAge:       defaultGCAge,
 		imageRecords: make(map[string]*imageRecord),
 		cadvisor:     cadvisorInterface,
 		recorder:     recorder,
 		nodeRef:      nodeRef,
-		initialized:  false,
 	}
 
 	return im, nil
 }
 
-func (im *realImageManager) Start() error {
-	go util.Until(func() {
-		// Initial detection make detected time "unknown" in the past.
-		var ts time.Time
-		if im.initialized {
-			ts = time.Now()
-		}
-		err := im.detectImages(ts)
```

Comment:
Why we need refresh kubelet's internal representation of images up to date? We get the docker's images file system information from cAdvisor, see https://github.com/kubernetes/kubernetes/pull/18151/files#diff-632dbe0678962cb022748a3dfdff09d8R116, which mean we don't need the internal representation of image to caculate disk usage information.
As for getting image usage information so that we can determine whether a image could be garbage collected, method `GarbageCollect()` wil call `freespace()`, which will then call `detectImages()` to collect image usage information https://github.com/kubernetes/kubernetes/pull/18151/files#diff-632dbe0678962cb022748a3dfdff09d8R116